// angular2-signaturepad/esm2020/lib/angular2-signaturepad.module.mjs
import { NgModule } from '@angular/core';
import { SignaturePad } from './angular2-signaturepad.component';
import * as i0 from "@angular/core";
export class SignaturePadModule {
}
SignaturePadModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: SignaturePadModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SignaturePadModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: SignaturePadModule, declarations: [SignaturePad], exports: [SignaturePad] });
SignaturePadModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: SignaturePadModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: SignaturePadModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SignaturePad],
                    imports: [],
                    exports: [SignaturePad],
                }]
        }] });
