// @metismenu/angular/esm2015/lib/metismenu-angular.component.js
import { Component, ElementRef, Input } from '@angular/core';
import MetisMenu from 'metismenujs';
export class MetismenuAngularComponent {
    constructor(elem) {
        this.elem = elem;
        this.toggle = 'true';
        this.parentTrigger = 'li';
        this.subMenu = 'ul';
        this.triggerElement = 'a';
    }
    ngAfterViewInit() {
        this.mmOpts = {
            toggle: (this.toggle.toLowerCase() === 'true'),
            parentTrigger: this.parentTrigger || 'li',
            subMenu: this.subMenu || 'ul',
            triggerElement: this.triggerElement || 'a',
        };
        this.mm = new MetisMenu(this.elem.nativeElement, this.mmOpts);
    }
    ngOnInit() { }
    ngOnDestroy() {
        this.mm.dispose();
    }
}
MetismenuAngularComponent.decorators = [
    { type: Component, args: [{
                selector: 'metis-menu, [metis-menu]',
                // inputs: ['toggle', 'parentTrigger', 'subMenu', 'triggerElement'],
                template: `<ng-content></ng-content>`
            },] }
];
MetismenuAngularComponent.ctorParameters = () => [
    { type: ElementRef }
];
MetismenuAngularComponent.propDecorators = {
    toggle: [{ type: Input }],
    parentTrigger: [{ type: Input }],
    subMenu: [{ type: Input }],
    triggerElement: [{ type: Input }]
};
