// @almothafar/angular-signature-pad/esm2020/lib/angular-signature-pad.module.mjs
import { NgModule } from '@angular/core';
import { SignaturePadComponent } from './angular-signature-pad.component';
import * as i0 from "@angular/core";
export class AngularSignaturePadModule {
}
AngularSignaturePadModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.0", ngImport: i0, type: AngularSignaturePadModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AngularSignaturePadModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.0", ngImport: i0, type: AngularSignaturePadModule, declarations: [SignaturePadComponent], exports: [SignaturePadComponent] });
AngularSignaturePadModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.0", ngImport: i0, type: AngularSignaturePadModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.0", ngImport: i0, type: AngularSignaturePadModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SignaturePadComponent],
                    imports: [],
                    exports: [SignaturePadComponent],
                }]
        }] });
